src/Tests/Tests/Shared/TestMath.gen.cs

```diff
@@ -1523,6 +1523,46 @@ public static void degrees_double4()
             TestUtils.AreEqual(double4(TestUtils.SignedDoubleQNaN(), double.PositiveInfinity, double.PositiveInfinity, double.PositiveInfinity), degrees(double4(TestUtils.SignedDoubleQNaN(), double.PositiveInfinity, double.PositiveInfinity, double.PositiveInfinity)), 1, false);
         }
 
+        [TestCompiler]
+        public static void sign_int()
+        {
+            TestUtils.AreEqual(-1, sign(-156));
+            TestUtils.AreEqual(-1, sign(-2147483468));
+            TestUtils.AreEqual(0, sign(0));
+            TestUtils.AreEqual(1, sign(2147483467));
+            TestUtils.AreEqual(1, sign(123));
+            TestUtils.AreEqual(1, sign(int.MaxValue));
+            TestUtils.AreEqual(-1, sign(int.MinValue));
+        }
+
+        [TestCompiler]
+        public static void sign_int2()
+        {
+            TestUtils.AreEqual(int2(0, 0), sign(int2(0, 0)));
+            TestUtils.AreEqual(int2(0, 1), sign(int2(0, 125)));
+            TestUtils.AreEqual(int2(1, 1), sign(int2(125, 2147477267)));
+            TestUtils.AreEqual(int2(0, -1), sign(int2(0, -1637)));
+            TestUtils.AreEqual(int2(-1, -1), sign(int2(-1387, -2147482398)));
+            TestUtils.AreEqual(int2(-1, 1), sign(int2(int.MinValue, int.MaxValue)));
+        }
+
+        [TestCompiler]
+        public static void sign_int3()
+        {
+            TestUtils.AreEqual(int3(0, 1, 1), sign(int3(0, 125, 2147477267)));
+            TestUtils.AreEqual(int3(1, 1, 1), sign(int3(297, 439, 2147477)));
+            TestUtils.AreEqual(int3(-1, 0, 1), sign(int3(-1637, 0, 2147231337)));
+            TestUtils.AreEqual(int3(-1, -1, -1), sign(int3(-34597, -1637, -2147174350)));
+            TestUtils.AreEqual(int3(-1, -1, 1), sign(int3(int.MinValue, -8183, int.MaxValue)));
+        }
+
+        [TestCompiler]
+        public static void sign_int4()
+        {
+            TestUtils.AreEqual(int4(-1, -1, 0, 1), sign(int4(-3899, -97268, 0, 2097152)));
+            TestUtils.AreEqual(int4(1, -1, 0, 1), sign(int4(int.MaxValue, int.MinValue, 0, 39876539)));
```

Comment:
We need to move these tests to `TestMath.cs` instead of where it is now in `TestMath.gen.cs`. Any file with "gen" in the name is generated and your changes will be lost the next time someone runs the code generator.

Alternatively, you could generate these tests with the code generator but I don't think that's necessary to do right now.

Reply:
ack, good catch

Reply:
Thanks for pointing it out!  I will make the changes.